src/routes/Details/Details.tsx

```diff
@@ -27,43 +30,82 @@ export default function Details() {
+        {isLoading && (
+          <div className="flex h-full items-center justify-center">
+            <Spinner />
+          </div>
+        )}
+        {isError && (
+          <div className="flex h-full items-center justify-center">
+            <p className="font-bold text-red-500">Error occurred</p>
+          </div>
+        )}
+        {data && (
+          <>
+            <img
+              src={data.image}
+              alt={data.name}
+              className="mb-4 w-full rounded"
+            />
+            <ul className="space-y-2 dark:text-white">
+              <li className="text-xl">
+                <strong>Galactic Registry — Citadel of Ricks</strong>
+              </li>
+              <li>
+                <strong>Subject: </strong> {data.name}
+              </li>
+              <li>
+                <strong>Species: </strong> {data.species}
+              </li>
+              <li>
+                <strong>Gender: </strong> {data.gender}
+              </li>
+              <li>
+                <strong>Status: </strong> {data.status}
+              </li>
+              <li>
+                <strong>Last Known Location: </strong>{' '}
+                {data.location.name !== 'unknown'
+                  ? data.location.name
+                  : 'Location data is currently unavailable.'}
+              </li>
+              <li>
+                <strong>Origin: </strong>{' '}
+                {data.origin.name !== 'unknown'
+                  ? data.origin.name
+                  : 'Origin details remain unverified or classified.'}
+              </li>
+              <li>
+                <strong>Notes:</strong>{' '}
+                {`Subject exhibits typical behavior for species ${data.species}, with no exceptional traits reported. Current status ${data.status !== 'unknown' ? 'confirmed' : 'currently unverified or unknown'}, though subject's survival beyond this report is subject to cosmic probability and/or Rick-level chaos.`}
+              </li>
+              <li>
+                {data.origin.name !== 'unknown' ? (
+                  <>
+                    Origin details have been recorded as{' '}
+                    <strong>{data.origin.name}</strong>.
+                  </>
+                ) : (
+                  <>Origin details remain unverified or classified.</>
+                )}
+              </li>
+              <li>
+                {data.location.name !== 'unknown' ? (
+                  <>
+                    Last known location is documented as{' '}
+                    <strong>{data.location.name}</strong>.
+                  </>
+                ) : (
+                  <>Location data is currently unavailable.</>
+                )}
+              </li>
+              <li>Monitoring continues.</li>
+            </ul>
+          </>
+        )}
+        <div className="flex items-center justify-center pt-10">
+          <Button onClick={() => refetch()}>Refresh</Button>
+        </div>
```

Comment:
```suggestion
        <div className="flex items-center justify-center pt-10">
          <Button onClick={() => refetch()}>Refresh ↻</Button>
        </div>
```
Let's keep up the traditions :)